src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,200 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.exceptions.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+    @Override
+    public Optional<ShoppingCart> getUserShoppingCart(Long userId) {
+        String query = "SELECT * FROM shopping_carts WHERE user_id = ? AND deleted = FALSE";
+        ShoppingCart cart = null;
```

Comment:
Not nice to have such declaration. As an option - have another method to extract shop-cart, return here this cart and then call method to populate with products

Reply:
Menthor told nothing about this so I leave my version :)